core/src/main/java/com/oath/oak/OakMap.java

```diff
@@ -252,9 +250,33 @@ public void putIfAbsentComputeIfPresent(K key, V value, Consumer<ByteBuffer> com
         internalOakMap.putIfAbsentComputeIfPresent(key, value, computer);
     }
 
+
+    /* ---------------- NavigableMap API methods -------------- */
+
+    @Override
+    public Entry<K, V> lowerEntry(K key) {
+        if (this.isSubmap()) throw new UnsupportedOperationException();
+
+        if (key == null) throw new NullPointerException();
+
+        return internalOakMap.lowerEntry(key);
+    }
+
+    @Override
+    public K lowerKey(K key) {
+        if (this.isSubmap()) throw new UnsupportedOperationException();
+
+        if (key == null) throw new NullPointerException();
+
+        return (K) internalOakMap.lowerEntry(key).getKey();
+    }
+
+
+    /* ---------------- ConcurrentNavigableMap API methods -------------- */
+
     /*-------------- SubMap --------------*/
-    // package visibility to be used by the views
-    public boolean inBounds(K key) {
+
+    boolean inBounds(Object key) {
```

Comment:
why changed to Object and not Key?

Reply:
The standard Map interfaces take Object arguments (to preserve the substitution principle) and it trickled down here